EapWebInterface/src/main/java/de/hpi/unicorn/application/pages/input/generator/DependenciesPanel.java

```diff
@@ -303,24 +336,79 @@ private void addListOfDependencies() {
         {
             @Override
             protected Object load() {
-                return new ArrayList<String>(dependenciesInput.keySet());
+                ArrayList<String> keys = new ArrayList<>(dependenciesInput.keySet());
+                dependencyValues = new ArrayList<>();
+                for (String key : keys) {
+                    dependencyValues.add(new DependencyInput(key));
+                }
+                return dependencyValues;
             }
         };
+        listContainer = new WebMarkupContainer("dependenciesContainer");
+        listContainer.setOutputMarkupId(true);
+        Model selectAllModel = new Model<Boolean>() {
+            @Override
+            public Boolean getObject() {
+                return panel.allSelected;
+            }
+            @Override
+            public void setObject(Boolean bool) {
+                panel.allSelected = bool;
+            }
+        };
+        listContainer.add(new AjaxCheckBox("selectAllCheckbox", selectAllModel) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') This class has 8 parents which is greater than 5 authorized. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://bpt-lab.org/sonarqube/coding_rules#rule_key=squid%3AMaximumInheritanceDepth)